hardware/national_instruments_x_series.py

```diff
@@ -1655,22 +1716,48 @@ def count_odmr(self, length=100):
         if len(self._scanner_counter_daq_tasks) < 1:
             self.log.error(
                 'No counter is running, cannot scan an ODMR line without one.')
-            return np.array([-1.])
+            return True, np.array([-1.])
 
         if len(self._scanner_ai_channels) > 0 and self._scanner_analog_daq_task is None:
-            self.log.error('No analog is running, cannot do ODMR without one.')
-            return np.array([-1.])
+            self.log.error('No analog task is running, cannot do ODMR without one.')
+            return True, np.array([-1.])
 
         # check if length setup is correct, if not, adjust.
-        self.set_odmr_length(length)
+        odmr_length_to_set = length * self._oversampling * 2 if self._odmr_pulser_daq_task else length
+        if self.set_odmr_length(odmr_length_to_set) < 0:
+            self.log.error('An error arose while setting the odmr lenth to {}.'.format(odmr_length_to_set))
+            return True, np.array([-1.])
+
         try:
             # start the scanner counting task that acquires counts synchroneously
             daq.DAQmxStartTask(self._scanner_counter_daq_tasks[0])
             if len(self._scanner_ai_channels) > 0:
                 daq.DAQmxStartTask(self._scanner_analog_daq_task)
         except:
             self.log.exception('Cannot start ODMR counter.')
-            return np.array([-1.])
+            return True, np.array([-1.])
+
+        try:
+            if self._odmr_pulser_daq_task:
+                pulse_pattern = np.zeros(self._oversampling*2, dtype=np.uint32)
```

Comment:
Maybe a comment about what this pattern is doing?

Reply:
done.